### PR TITLE
refactor(cli): enhance the plugin add command to take input argument

### PR DESCRIPTION
**Pull request by UnleashedMind**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli/src/commands/plugin/add.ts [hunk 1]
```diff
@@ -1,19 +1,110 @@
...
+  if (candicatePluginDirPaths.length === 0) {
+    context.print.error('Can not locate the plugin package.');
+    result = await promptForPluginPath();
+  } else if (candicatePluginDirPaths.length === 1) {
+    context.print.green('Plugin package found.');
+    context.print.blue(candicatePluginDirPaths[0]);
+    const { confirmed } = await inquirer.prompt({
+      type: 'confirm',
+      name: 'confirmed',
+      message: `Please confirm to add the plugin package to your Amplify CLI.`,
```
Comment (attilah) [inline packages/amplify-cli/src/commands/plugin/add.ts]: Based on our guide we should not use "Please" and similar

File: packages/amplify-cli/src/commands/plugin/add.ts [hunk 2]
```diff
@@ -1,19 +1,110 @@
...
+      default: true,
+    });
+    if (confirmed) {
+      result = candicatePluginDirPaths[0];
+    }
+  } else if (candicatePluginDirPaths.length > 1) {
+    context.print.warning('Multiple plugins with the package name are found.');
```
Comment (attilah) [inline packages/amplify-cli/src/commands/plugin/add.ts]: We should output what packages is a duplicate: "Multiple plugins found with the same package name: {packageName}".

File: packages/amplify-cli/src/commands/plugin/add.ts [hunk 3]
```diff
@@ -1,19 +1,110 @@
...
 import {
     addUserPluginPackage,
     addExcludedPluginPackage as addFromExcluded,
     confirmAndScan,
 } from '../../plugin-manager';
 import inquirer, { InquirerOption, EXPAND } from '../../domain/inquirer-helper';
 import { AddPluginError } from '../../domain/add-plugin-result';
+import { normalizePluginDirectory } from '../../plugin-helpers/scan-plugin-platform';
 
 const NEWPLUGINPACKAGE = 'A new plugin package';
```
[the remark this answers is not in the document]
Comment (UnleashedMind) [reply packages/amplify-cli/src/commands/plugin/add.ts]: check

File: packages/amplify-cli/src/commands/plugin/add.ts [hunk 4]
```diff
@@ -1,19 +1,110 @@
...
-
+const CANCAL = 'cancel';
```
[the remark this answers is not in the document]
Comment (UnleashedMind) [reply packages/amplify-cli/src/commands/plugin/add.ts]: check

File: packages/amplify-cli/src/commands/plugin/add.ts (hunk 1 above)
Comment (UnleashedMind) [reply packages/amplify-cli/src/commands/plugin/add.ts]: check

File: packages/amplify-cli/src/commands/plugin/add.ts [hunk 5]
```diff
@@ -1,19 +1,110 @@
...
 
 export async function run(context: Context) {
+  if (context.input.subCommands && context.input.subCommands.length > 1) {
+    const input = context.input.subCommands[1];
+    const { excluded } = context.pluginPlatform;
+    if (excluded[input] && excluded[input].length > 0) {
+      const { confirmed } = await inquirer.prompt({
+        type: 'confirm',
+        name: 'confirmed',
+        message: `Add from previously removed ${input} plugin`,
```
[the remark this answers is not in the document]
Comment (UnleashedMind) [reply packages/amplify-cli/src/commands/plugin/add.ts]: the command that the user ran was actually `amplify plugin remove`, so for the user it's `remove`, and we only put it inside the excluded array if the package is under the scan radar.

File: packages/amplify-cli/src/commands/plugin/add.ts (hunk 2 above)
Comment (UnleashedMind) [reply packages/amplify-cli/src/commands/plugin/add.ts]: the package name is the user's input. and below this line is the selection with all the package paths prompted to the user

**Pull request merged**